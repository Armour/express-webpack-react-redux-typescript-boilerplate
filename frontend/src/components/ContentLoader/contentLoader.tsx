import React from 'react';
import ContentLoader from 'react-content-loader';

const styles = require('./contentLoader.scss');

export const BodyContentLoader = () => (
  <div className={styles.bodyContentLoader}>
    <ContentLoader
      height={200}
      width={400}
      speed={1}
      primaryColor='#f3f3f3'
      secondaryColor='#d7d6d8'
    >
      <rect x='160.5' y='11' rx='4' ry='4' width='76.63' height='15.1' />
      <rect x='60.5' y='35.05' rx='1' ry='1' width='275.4' height='151.36' />
    </ContentLoader>
  </div>
);

export const HeaderContentLoader = () => (
  <ContentLoader
    height={95}
    width={2500}
    speed={1}
    primaryColor='#f3f3f3'
    secondaryColor='#d7d6d8'
  >
    <rect x='0' y='0' width='2500' height='95' />
  </ContentLoader>
);

export const FooterContentLoader = () => (
  <ContentLoader
    height={400}
    width={2500}
    speed={1}
    primaryColor='#f3f3f3'
    secondaryColor='#d7d6d8'
  >
    <rect x='0' y='0' width='2500' height='400' />
  </ContentLoader>
);
