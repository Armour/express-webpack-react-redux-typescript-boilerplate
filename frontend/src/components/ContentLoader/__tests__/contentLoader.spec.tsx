import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { BodyContentLoader, FooterContentLoader, HeaderContentLoader } from '../contentLoader';

describe('HeaderContentLoader', () => {
  it('should renders correctly', () => {
    const renderer = ShallowRenderer.createRenderer();
    const result = renderer.render(
      <HeaderContentLoader />,
    );
    expect(result).toMatchSnapshot();
    renderer.unmount();
  });
});

describe('BodyContentLoader', () => {
  it('should renders correctly', () => {
    const renderer = ShallowRenderer.createRenderer();
    const result = renderer.render(
      <BodyContentLoader />,
    );
    expect(result).toMatchSnapshot();
    renderer.unmount();
  });
});

describe('FooterContentLoader', () => {
  it('should renders correctly', () => {
    const renderer = ShallowRenderer.createRenderer();
    const result = renderer.render(
      <FooterContentLoader />,
    );
    expect(result).toMatchSnapshot();
    renderer.unmount();
  });
});
