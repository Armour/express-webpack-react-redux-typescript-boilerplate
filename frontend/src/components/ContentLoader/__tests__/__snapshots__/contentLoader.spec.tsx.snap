// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyContentLoader should renders correctly 1`] = `
<div
  className="bodyContentLoader"
>
  <ContentLoader
    height={200}
    primaryColor="#f3f3f3"
    secondaryColor="#d7d6d8"
    speed={1}
    width={400}
  >
    <rect
      height="15.1"
      rx="4"
      ry="4"
      width="76.63"
      x="160.5"
      y="11"
    />
    <rect
      height="151.36"
      rx="1"
      ry="1"
      width="275.4"
      x="60.5"
      y="35.05"
    />
  </ContentLoader>
</div>
`;

exports[`FooterContentLoader should renders correctly 1`] = `
<ContentLoader
  height={400}
  primaryColor="#f3f3f3"
  secondaryColor="#d7d6d8"
  speed={1}
  width={2500}
>
  <rect
    height="400"
    width="2500"
    x="0"
    y="0"
  />
</ContentLoader>
`;

exports[`HeaderContentLoader should renders correctly 1`] = `
<ContentLoader
  height={95}
  primaryColor="#f3f3f3"
  secondaryColor="#d7d6d8"
  speed={1}
  width={2500}
>
  <rect
    height="95"
    width="2500"
    x="0"
    y="0"
  />
</ContentLoader>
`;
